File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/impl/GuiScreenImpl.java
package rbq.wtf.lycoris.client.wrapper.wrappers.impl;



public abstract class GuiScreenImpl{
    public abstract void drawScreen(int mouseX, int mouseY, float partialTicks);
    public void initGui(){}
    public void onGuiClosed(){}
    public boolean keyTyped(char typedChar, int keyCode){
        return true;
    }
    public void mouseClicked(int mouseX, int mouseY, int mouseButton){}
    public void mouseReleased(int mouseX, int mouseY, int state){}
    public void updateScreen(){}
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/impl/GuiScreenImpl.kt
package rbq.wtf.lycoris.client.wrapper.impl

abstract class GuiScreenImpl {
    abstract fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float)
    open fun initGui() {}
    open fun onGuiClosed() {}
    open fun keyTyped(typedChar: Char, keyCode: Int): Boolean = true
    open fun mouseClicked(mouseX: Int, mouseY: Int, mouseButton: Int) {}
    open fun mouseReleased(mouseX: Int, mouseY: Int, state: Int) {}
    open fun updateScreen() {}
}